function.tf

```diff
@@ -0,0 +1,78 @@
+locals {
+  enable_function = var.enable_function || var.enable_extensions
+}
+
+resource "google_service_account" "cloud_functions" {
+  account_id  = "${var.name}-func"
+  description = "A service account for the Observe Cloud Functions"
+}
+
+resource "google_project_iam_member" "cloud_functions" {
+  for_each = toset(var.function_roles)
```

Comment:
The variable itself should be `set(string)` rather than converting it.

Reply:
good point, thx